docs/sources/setup-grafana/configure-security/configure-authentication/gitlab/index.md

```diff
@@ -19,85 +19,83 @@ weight: 1000
 
-You'll get an _Application Id_ and a _Secret_ in return; we'll call them
-`GITLAB_APPLICATION_ID` and `GITLAB_SECRET` respectively for the rest of this
-section.
+1. Create an OAuth application in GitLab.
 
-## Enable GitLab in Grafana
+   1. Set the redirect URI to `http://<my_grafana_server_name_or_ip>:<grafana_server_port>/login/gitlab`.
 
-In this example, we'll assume you use the public `gitlab.com` instance, but you
-can use your own instance of GitLab instead by replacing `auth_url`, `token_url` with the URL of your instance.
+      Ensure that the Redirect URI is the complete HTTP address that you use to access Grafana via your browser, but with the appended path of `/login/gitlab`.
 
-You can find these URLs in the `well known` configuration file of your GitLab instance, for example `https://gitlab.com/.well-known/openid-configuration`.
+      For the Redirect URI to be correct, it might be necessary to set the `root_url` option in the `[server]`section of the Grafana configuration file. For example, if you are serving Grafana behind a proxy.
 
-Add the following to your Grafana configuration file to enable GitLab
-authentication:
+   1. Set the OAuth2 scopes to `openid`, `email` and `profile`.
 
-```bash
-[auth.gitlab]
-enabled = true
-allow_sign_up = true
-auto_login = false
-client_id = GITLAB_APPLICATION_ID
-client_secret = GITLAB_SECRET
-scopes = openid email profile
-auth_url = https://gitlab.com/oauth/authorize
-token_url = https://gitlab.com/oauth/token
-allowed_groups =
-role_attribute_path =
-role_attribute_strict = false
-allow_assign_grafana_admin = false
-tls_skip_verify_insecure = false
-tls_client_cert =
-tls_client_key =
-tls_client_ca =
-use_pkce = true
-```
+1. Refer to the following table to update field values located in the `[auth.gitlab]` section of the Grafana configuration file:
+
+   | Field                        | Description                                                                                  |
+   | ---------------------------- | -------------------------------------------------------------------------------------------- |
+   | `client_id`, `client_secret` | These values must match the client ID and client secret from your GitLab OAuth2 application. |
+   | `enabled`                    | Enables GitLab authentication. Set this value to `true`.                                     |
+
+   Review the list of other GitLab [configuration options]({{< relref "#configuration-options" >}}) and complete them, as necessary.
+
+1. Optional: [Configure a refresh token]({{< relref "#configure-a-refresh-token" >}}):
 
-You may have to set the `root_url` option of `[server]` for the callback URL to be
-correct. For example in case you are serving Grafana behind a proxy.
+   a. Enable `accessTokenExpirationCheck` feature toggle.
 
-Restart the Grafana backend for your changes to take effect.
+   c. Set `use_refresh_token` to `true` in `[auth.generic_oauth]` section in Grafana configuration file.
```

Comment:
what happened to `b`?

Reply:
Hehe, well spotted! Got swallowed by `c` 🙈